packages/@aws-cdk/aws-config/lib/rule.ts

```diff
@@ -255,12 +260,14 @@ export class ManagedRule extends RuleNew {
       physicalName: props.configRuleName,
     });
 
+    this.scope = props.scope;
+
     const rule = new CfnConfigRule(this, 'Resource', {
       configRuleName: this.physicalName,
       description: props.description,
       inputParameters: props.inputParameters,
       maximumExecutionFrequency: props.maximumExecutionFrequency,
-      scope: Lazy.anyValue({ produce: () => this.scope }),
+      scope: Lazy.anyValue({ produce: () => renderScope(this.scope) }),
```

Comment:
This doesn't need to be `Lazy` anymore.

... unless, we want to retain and deprecate the existing methods (e.g., `fromResourceType`), but allow their use as long as the scope wasn't set in the constructor props.

Reply:
i'd originally dropped the `Lazy`, but ended up keeping it as there's usage where the property is set outside of the supoer call -

i.e. the `CloudFormationStackDriftDetectionCheck` managed rule which uses a resource identifier of it's own `stack id` with the value of `props.ownStackOnly ? Stack.of(this).stackId`

Reply:
Yeah, I missed the implication of the `CloudFormationStackDriftDetectionCheck`. I guess this has to stay. Maybe a quick comment explaining the Lazy since it's non-obvious?

Reply:
sounds good!